sonar-helm-for-iac/converters/data-loading.go

```diff
@@ -8,39 +8,43 @@ import (
 
 // TemplateSources contains all the sources needed to evaluate a template
 type TemplateSources struct {
-	Name                  string
-	RawTemplate           string
-	fileNameToFileContent map[string]string
+	Name  string
+	files Files
 }
 
-func NewTemplateSources(name string, rawTemplate string, fileNameToFileContent map[string]string) *TemplateSources {
-	return &TemplateSources{name, rawTemplate, fileNameToFileContent}
+func NewTemplateSources(name string, fileNameToFileContent Files) *TemplateSources {
+	return &TemplateSources{name, fileNameToFileContent}
 }
 
-// NumAdditionalSources returns the number of sources required for this template.
-func (ts *TemplateSources) NumAdditionalSources() int {
-	return len(ts.fileNameToFileContent)
+// NumSources returns the number of sources required for evaluation of this template (including itself).
+func (ts *TemplateSources) NumSources() int {
+	return len(ts.files)
 }
 
-func (ts *TemplateSources) SourceFile(name string) (string, error) {
-	if _, ok := ts.fileNameToFileContent[name]; !ok {
-		return "", fmt.Errorf("source file %s not found", name)
+func (ts *TemplateSources) SourceFile(name string) ([]byte, error) {
```

Comment:
This can be changed to the private method as all calls are later are used later as `string`